test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -19,263 +19,103 @@
-     */
-    public static final String REST_TESTS_BLACKLIST = "tests.rest.blacklist";
-    /**
-     * Property that allows to control whether spec validation is enabled or not (default true).
+     * Convert the entity from a {@link Response} into a map of maps.
      */
-    public static final String REST_TESTS_VALIDATE_SPEC = "tests.rest.validate_spec";
+    public static Map<String, Object> entityAsMap(Response response) throws IOException {
+        XContentType xContentType = XContentType.fromMediaTypeOrFormat(response.getEntity().getContentType().getValue());
+        try (XContentParser parser = xContentType.xContent().createParser(response.getEntity().getContent())) {
+            return parser.map();
+        }
+    }
+
+    private final List<HttpHost> clusterHosts;
     /**
-     * Property that allows to control where the REST spec files need to be loaded from
+     * A client for the running Elasticsearch cluster. Lazily initialized on first use.
      */
-    public static final String REST_TESTS_SPEC = "tests.rest.spec";
-
-    public static final String REST_LOAD_PACKAGED_TESTS = "tests.rest.load_packaged";
-
-    private static final String DEFAULT_TESTS_PATH = "/rest-api-spec/test";
-    private static final String DEFAULT_SPEC_PATH = "/rest-api-spec/api";
-
+    private volatile RestClient client;
     /**
-     * This separator pattern matches ',' except it is preceded by a '\'.
-     * This allows us to support ',' within paths when it is escaped with a slash.
-     *
-     * For example, the path string "/a/b/c\,d/e/f,/foo/bar,/baz" is separated to "/a/b/c\,d/e/f", "/foo/bar" and "/baz".
-     *
-     * For reference, this regular expression feature is known as zero-width negative look-behind.
-     *
+     * A client for the running Elasticsearch cluster configured to take test administrative actions like remove all indexes after the test
+     * completes. Lazily initialized on first use.
      */
-    private static final String PATHS_SEPARATOR = "(?<!\\\\),";
-
-    private final List<BlacklistedPathPatternMatcher> blacklistPathMatchers = new ArrayList<>();
-    private final URL[] clusterUrls;
-    private static RestTestExecutionContext restTestExecutionContext;
-    private static RestTestExecutionContext adminExecutionContext;
-
-    private final RestTestCandidate testCandidate;
+    private volatile RestClient adminClient;
 
-    public ESRestTestCase(RestTestCandidate testCandidate) {
-        this.testCandidate = testCandidate;
-        String[] blacklist = resolvePathsProperty(REST_TESTS_BLACKLIST, null);
-        for (String entry : blacklist) {
-            this.blacklistPathMatchers.add(new BlacklistedPathPatternMatcher(entry));
-        }
+    public ESRestTestCase() {
         String cluster = System.getProperty("tests.rest.cluster");
         if (cluster == null) {
             throw new RuntimeException("Must specify tests.rest.cluster for rest tests");
```

Comment:
Sounds like a good idea? I'm not sure what we should change it too. These are still "rest" tests so I think it is ok? I'm not sure.